pintracker/statelesstracker/statelesstracker.go

```diff
@@ -0,0 +1,510 @@
+package stateless
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/ipfs/ipfs-cluster/api"
+	"github.com/ipfs/ipfs-cluster/optracker"
+	"github.com/ipfs/ipfs-cluster/pintracker/util"
+
+	rpc "github.com/hsanjuan/go-libp2p-gorpc"
+	cid "github.com/ipfs/go-cid"
+	logging "github.com/ipfs/go-log"
+	peer "github.com/libp2p/go-libp2p-peer"
+)
+
+var logger = logging.Logger("pintracker")
+
+// Tracker uses the optracker.OperationTracker to manage
+// transitioning shared ipfs-cluster state (Pins) to the local IPFS node.
+type Tracker struct {
+	config *Config
+
+	optracker *optracker.OperationTracker
+
+	opErrsMu sync.RWMutex
+	opErrs   map[string]api.PinInfo
+
+	remotesMu sync.RWMutex
+	remotes   map[string]api.PinInfo
```

Comment:
We shouldn't store `remotes` as it can potentially grow very big (in the same order as the original state that we are trying to remove)

This has consequences (cannot distinguish between something `unpinned` or `remote`). Perhaps this is fine.